Code/Framework/AzToolsFramework/Tests/Input/QtEventToAzInputMapperTests.cpp

```diff
@@ -0,0 +1,535 @@
+                [this]([[maybe_unused]] const AzFramework::InputChannel* inputChannel, QEvent* event)
+                {
+                    const QEvent::Type eventType = event->type();
+
+                    if (eventType == QEvent::Type::MouseButtonPress ||
+                        eventType == QEvent::Type::MouseButtonRelease ||
+                        eventType == QEvent::Type::MouseButtonDblClick)
+                    {
+                        m_signalEvents.push_back(QtEventInfo(static_cast<QMouseEvent*>(event)));
+                        event->accept();
+                    }
+                    else if (eventType == QEvent::Type::Wheel)
+                    {
+                        m_signalEvents.push_back(QtEventInfo(static_cast<QWheelEvent*>(event)));
+                        event->accept();
+                    }
+                    else if (eventType == QEvent::Type::KeyPress ||
+                            eventType == QEvent::Type::KeyRelease ||
+                            eventType == QEvent::Type::ShortcutOverride)
+                    {
+                        m_signalEvents.push_back(QtEventInfo(static_cast<QKeyEvent*>(event)));
+                        event->accept();
+                    }
+                });
+        }
+
+        void TearDown() override
+        {
+            m_inputChannelMapper.reset();
+
+            m_rootWidget.reset();
+
+            AllocatorsTestFixture::TearDown();
+        }
+
+        void OnInputChannelEvent(const AzFramework::InputChannel& inputChannel, bool& hasBeenConsumed) override
+        {
+            ASSERT_FALSE(hasBeenConsumed);
+
+            const AzFramework::InputChannelId& inputChannelId = inputChannel.GetInputChannelId();
+            const AzFramework::InputDeviceId& inputDeviceId = inputChannel.GetInputDevice().GetInputDeviceId();
+
+            if (AzFramework::InputDeviceMouse::IsMouseDevice(inputDeviceId))
+            {
+                if (IsMouseButton(inputChannelId))
+                {
+                    m_azChannelEvents.push_back(AzEventInfo(inputChannel));
+                    hasBeenConsumed = m_captureAzEvents;
+                }
+                else if (inputChannelId == AzFramework::InputDeviceMouse::Movement::Z)
+                {
+                    m_azChannelEvents.push_back(AzEventInfo(inputChannel));
+                    hasBeenConsumed = m_captureAzEvents;
+                }
+            }
+            else if (AzFramework::InputDeviceKeyboard::IsKeyboardDevice(inputDeviceId))
+            {
+                m_azChannelEvents.push_back(AzEventInfo(inputChannel));
+                hasBeenConsumed = m_captureAzEvents;
+            }
+        }
+
+        void OnInputTextEvent(const AZStd::string& textUTF8, bool& hasBeenConsumed) override
+        {
+            ASSERT_FALSE(hasBeenConsumed);
+
+            m_azTextEvents.push_back(textUTF8);
+            hasBeenConsumed = m_captureTextEvents;
+        }
+
+        struct QtEventInfo
```

Comment:
suggestion: Comment here detailing usage could be helpful (and again could be moved out to a util class in future).